newton.go

```diff
@@ -0,0 +1,143 @@
+// Copyright ©2015 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package optimize
+
+import (
+	"math"
+
+	"github.com/gonum/floats"
+	"github.com/gonum/matrix/mat64"
+)
+
+// Newton implements a modified Newton's method for Hessian-based unconstrained
+// minimization. It applies regularization when the Hessian is not positive
+// definite, and uses line-searches so it can converge to a local minimum from
```

Comment:
I'm not completely sure about this, but I think that for the theory a line search (together with some kind of Hessian modification) is necessary to get convergence from any starting point. The wording tried to reflect that.

I removed "uses line-searches so".